### PR TITLE
Change warning to DEBUG_MODE in mem_cache

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Modify the log message in `mem_cache` to `debug` at the request of the team.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/data/mem_cache.py [hunk 1]
```diff
@@ -168,8 +169,9 @@ def put(
 
                 if new_page > self.mem_size:
                     self.freeze()
-                    msg = "Memory pool reaches it's limit. Cannot cache more. Freeze it."
-                    logger.warning(msg)
+                    if os.getenv("DEBUG_MODE", "false").lower() == "true":
+                        msg = "Memory pool reaches it's limit. Cannot cache more. Freeze it."
+                        logger.warning(msg)
```
Comment (eunwoosh) [inline src/otx/core/data/mem_cache.py]:
```suggestion
                    logger.debug(msg)
```
How about using logger.debug?

Comment (harimkang) [reply src/otx/core/data/mem_cache.py]: Thanks! fixed - https://github.com/openvinotoolkit/training_extensions/pull/3757/commits/2ebfb98bf50221841a18eb7d0dcb306c8707145b

Comment (eunwoosh) [reply src/otx/core/data/mem_cache.py]: ah sorry I missed `msg`. Could you add it again?

Comment (harimkang) [reply src/otx/core/data/mem_cache.py]: Thanks, done - https://github.com/openvinotoolkit/training_extensions/pull/3757/commits/99fe5fc73076b2cd435c24e54dc694791aa3b489

**Review by eunwoosh — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]

**Pull request merged**